### PR TITLE
tutorial completed for flutter speech recognition using alan voice

### LINKED ISSUE #5933: [language]Flutter speech recognition using Alan voice
### Topic Suggestion
[language]Flutter speech recognition using Alan voice
 
#### Proposed article introduction
This is a tutorial session on flutter that teaches the reader how to implement speech recognition on your flutter application. By speaking to our application we can perform actions based on preset commands. When the application running on our devices captures a speech match we can perform any operation we want as we would any other function.

#### Key takeaways 
The reader will learn how to implement speech recognition in his application.
The reader will learn how to integrate voice assistance using commands defined.

#### Article quality
This tutorial session will show an in-depth example of implementing speech recognition with Alan's voice. 

#### References
https://pub.dev/packages/alan_voice
https://github.com/alan-ai/alan-sdk-flutter

### PR DESCRIPTION
**Pull request by Eze4Manuel**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #5933

### CODE REVIEW

**Comment by ahmadmardeni1** [comment]
Why did you delete the template? I've edited your initial comment. Please check what matches your article. @Eze4Manuel

**Comment by Eze4Manuel** [comment]
@ahmadmardeni1  
I have checked all that matches my article. Thnaks

**Comment by Eze4Manuel** [comment]
Hello Ahmad,
I have checked and updated the matches as you requested. Please check and
let me know of any changes or updates needed to my tutorial.

Thank you.

**Comment by ahmadmardeni1** [comment]
Plagiarism check done ✅

File: content/articles/flutter_speech_recognition_using_alan_voice/index.md [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+
+### Goals
+We will learn how to setup speech recognition using Alan voice plugin in our flutter application
+See how to set commands and call them in our application inorder to perform some operations.
+
+### Setting up our application
+Head over to your flutter application IDE and spin up a new flutter application and name it any suitable name of your choosing.
+Once your application is set and ready to run we need to integrate flutter alan voice package into our application, this can be obtained from [Flutter package alan_voice](https://pub.dev/packages/alan_voice). Get the latest version of the package and add it to your pubspec.yml file. Ensure the minimumSdkVersion for your application is 21 to check this you can update this in your android/app/build.gradle file.
+
+![set-up-application](/flutter_speech_recognition_using_alan_voice/application.png)
```
Comment (Jethro-magaji) [inline content/articles/flutter_speech_recognition_using_alan_voice/index.md:36]: Please format all your image path like this ! [starting] (/engineering-education/create-a-registration-form-using-react.js-and-connecting-it-with-firebase/start.PNG)

File: content/articles/flutter_speech_recognition_using_alan_voice/index.md [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+
+To achieve this we will first set up our flutter application, then we will link it to alan voice by integrating alan voice plugin into the application. Next we will test our speech recognition application. Finally we will create/edit/update our commands without updating our application.
+
+
+### Prerequisite
```
Comment (Jethro-magaji) [inline content/articles/flutter_speech_recognition_using_alan_voice/index.md:14]: Please put all your prerequisite in an unordered list format, also include other prerequisites necessary for the reader to follow along such as a preferable IDE, Flutter SDK version, Alan SDK etc.

File: content/articles/flutter_speech_recognition_using_alan_voice/index.md [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+![create_account](/flutter_speech_recognition_using_alan_voice/account.png)
+
+
+Inorder to get our Alan SDK key click on the integration button on the top navigation on the window. Select the platform you want to integrate to ours is android so we select android. Note that for and ios application we need to obtain the configuration for ios by selecting the IOS option
+
+
+### Running our application
+We import our plugin file into our application and we have in our main.dart file the following code
+
+```dart
```
Comment (Jethro-magaji) [inline content/articles/flutter_speech_recognition_using_alan_voice/index.md:53]: Please properly explain all your code below each block and add an in-line comment if necessary.

File: content/articles/flutter_speech_recognition_using_alan_voice/index.md [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+       );
+     }
+     debugPrint("got new command ${command}");
+     debugPrint("got new command ${command.data['command']}");
+   });
+ }
+```
+
+If our command registered on our account is matched we should call the LaunchApp.openApp function to open our youtube application. If we should run our app once more, click the speech recognition button and say open youtube once more our youtube application should open.
+
```
Comment (Jethro-magaji) [inline content/articles/flutter_speech_recognition_using_alan_voice/index.md:219]: Add a link to the complete code on GitHub.

**Review by Jethro-magaji — CHANGES_REQUESTED** [review]
Hi! @Eze4Manuel 
Well done for such an amazing article. However, there are some corrections you need to make.
Please, run your article on tools like Grammarly and ProWritingAid for better editing. This would save the review time to review your work.
Here's a [link](https://github.com/section-engineering-education/engineering-education/pull/6210/commits/acc44893d4ec258f4e077b9ce8e3e2ca2b9d6a9f) to a PR with little or no revision to guide you on formatting. However, if you have any questions or need help, you can reach out to me on the EngEd Slack channel.
Thanks.

**Comment by Eze4Manuel** [comment]
Hello Jethro, I have made all the updates and adjustments you highlighted.

**Comment by hectorkambow** [comment]
> Hello Jethro, I have made all the updates and adjustments you highlighted.

@Eze4Manuel have you gone through the entire article to check for other mistakes? Our peer reviewers are not expected to POINT OUT everything. 👍

**Review by Jethro-magaji — APPROVED** [review]
Hi! @Eze4Manuel 
Well done for coming up with such an amazing article. Please take note of these corrections to minimize revision next time.
Thanks

**Comment by WanjaMIKE** [comment]
Merged #6628 into master

**Pull request merged**